[Estuary] Update info dialog details

Hitcher opened the pull request:
## Description
<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here. -->
Introduces a metadata dialog when pressing Enter/OK on the metadata area of the video info dialog.
Works for all video types (movie, set, version, show, season, etc).
Adds IMDb top 250 label.
Only displays the version type if it's not the standard version.
Also fixes a few other issues with the video info dialog and text viewer dialog.

## Test build
Uses a different addon ID for easier testing.
[skin.estuary.test.zip](https://github.com/user-attachments/files/19938790/skin.estuary.test.zip)

## Motivation and context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
As discussed on Slack, makes the metadata behave the same as the plot (ie opens a separate dialog for viewing).
Removes the need to fix an issue where 2 buttons could be focused at once.
Fixes an issue with the plot text getting cut off at the bottom in the video info dialog.
Fixes an issue with the text getting cut off at the bottom in the text viewer dialog.
Fixes an issue where the indicator labels at the bottom of the dialog got moved when https://github.com/xbmc/xbmc/pull/26206 realigned the media flags.

## Further work
If approved will work on updating all the other info dialogs (ie album, artist, song, pvr, radio, and addon).

## How has this been tested?
<!--- Please describe in detail how you tested your change. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
Thoroughly tested across all areas where the video info dialog can be triggered (ie library, files, home, search, and favourites).

## Screenshots (if appropriate):

![screenshot00000](https://github.com/user-attachments/assets/49e01d46-9b15-4558-a444-d17c8c1a09ed)
Button focus problem.

![screenshot00001](https://github.com/user-attachments/assets/bd637f13-ae23-4da4-9830-92128d909ae4)
Misaligned indicator label problem.

![screenshot00002](https://github.com/user-attachments/assets/c6b74b73-0f96-41ce-836e-efd90c923792)
Realigned indicator labels.

![screenshot00013](https://github.com/user-attachments/assets/f995bc09-4bb6-4eec-a893-dfdf18d13ae2)
Plot text cut off problem.

![screenshot00014](https://github.com/user-attachments/assets/58b00cbd-ebfc-4fc4-a1cd-fdbc1956be9f)
Fixed plot text.

![screenshot00011](https://github.com/user-attachments/assets/25084b44-228a-42e1-a842-00ec8658ecb4)
Text viewer text cut off problem.

![screenshot00012](https://github.com/user-attachments/assets/41d3733b-5df3-475c-88ef-4ddd161c3b22)
Fixed text viewer.

![screenshot00003](https://github.com/user-attachments/assets/e1c472d1-2866-4137-8152-062890c7f5a0)
Metadata area focus with new indicator label.

![screenshot00004](https://github.com/user-attachments/assets/06f4412f-5223-4df7-b172-12cb8d40930d)
New dialog with full metadata details.

![screenshot00005](https://github.com/user-attachments/assets/96a92a5f-981b-479c-a8e5-dee979e3323f)
Movie set.

![screenshot00006](https://github.com/user-attachments/assets/405a5910-4a1d-4882-8754-66bfcd10c4c4)
Version.

![screenshot00008](https://github.com/user-attachments/assets/3e273810-5886-44fc-9488-809b84f97f17)
TV show.

![screenshot00009](https://github.com/user-attachments/assets/7dafdedb-3416-42e1-a4d6-bf7607e31f27)
Season.

![screenshot00010](https://github.com/user-attachments/assets/da0a6594-a0ee-4f4c-b940-ba17b86355df)
Episode.

![screenshot00007](https://github.com/user-attachments/assets/5cab85c0-a816-43db-b057-0c4a8c530b03)
IMDb Top 250 label.

## Types of change

<!--- What type of change does your code introduce? Put an `x` with no space in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` with no space in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

Hitcher committed:
Update info dialog details

jjd-uk commented:
Intial impression is that it looks good to me.

As you've introduced a new dialog for the metadata we don't have the same concern about the list getting too long, so personally I'd like to be **Path** added using `$INFO[ListItem.DecodedFileNameAndPath]` while retaining the path also shown when Refresh is highlighted so as to not confuse users used to finding it there.

jjd-uk commented:
One thing I'm not completely sure on is the font sizing.

The Plot textbox uses the default font13.

The old metadata panel on the right used font12, the new MetadataViewerVideoList and MetadataViewer now look to be using font13 which I guess is to match the font13 of the Plot textbox.

Then the TextViewer and MetadataViewer both use a larger font37, this seems odd to me that they use a larger font.

Personally would have changed the Plot textbox to font12 and kept the metadata details as font12, and perhaps also have TextViewer and  MetadataViewer use font12 as well.

KarellenX commented:
Thanks for doing this @Hitcher 

I definitely like these pop-ups with more room to show the metadata. 
When I first selected the metadata box, my immediate thought was "that's a big font", and I felt the metadata box looked big and cluttered. 
I changed to font12 as suggested by @jjd-uk and height to "40" and I felt it was more comfortable to view/read.

Strange that the plot uses the same large font, but it seems fine there. Maybe because of the increased number of elements on the metadata page that it looked cluttered with the bigger font.

Hitcher commented:
Thanks for the feedback. I'll try things with font12 when I get a chance.

Hitcher committed:
Merge branch 'xbmc:master' into update_info_dialog_details

Hitcher committed:
Update

Change font sizes
Add decoded file path

Hitcher commented:
Changed the font size and added decoded file path to the metadata details:

![screenshot00019](https://github.com/user-attachments/assets/4d157e74-e1bd-436e-9f6f-28f1b68bd468)

Changed the font size for the textviewer and metadata viewer dialogs:

![screenshot00021](https://github.com/user-attachments/assets/e6b486cd-22ba-4aa1-a76d-4eab9459dce1)

![screenshot00022](https://github.com/user-attachments/assets/1093b23c-01a2-4073-a735-a0d71838ddcf)

jjd-uk reviewed (approved):
LGTM

jjd-uk commented:
@Hitcher are you going to do a seperate PR for the other dialogs?

Please only merge by using **Squash and merge** or leave it for me to do.

![image](https://github.com/user-attachments/assets/04139281-04ce-4728-9efb-7fc7e0de2b57)

Hitcher commented:
This was only really for feedback ready for me to do all of them at once.

KarellenX commented:
@Hitcher 

I see this was for feedback. Will the change make it to v22?

ksooo commented:
Could we also have this for the pvr info dialog?

Hitcher commented:
> Could we also have this for the pvr info dialog?

Possibly.
